### PR TITLE
Added UI thread checks to InfoBarManager

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Relates to #1888

### CODE REVIEW

File: src/Integration.Vsix/InfoBar/InfoBarManager.cs [hunk 1]
```diff
@@ -70,26 +70,21 @@ public IInfoBar AttachInfoBar(Guid toolWindowGuid, string message, SonarLintImag
             return AttachInfoBarImpl(toolWindowGuid, message, null, imageMoniker);
         }
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability",
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/InfoBar/InfoBarManager.cs:73]: Miscellaneous: code smell said this wasn't needed

File: src/Integration.Vsix/InfoBar/InfoBarManager.cs [hunk 2]
```diff
@@ -70,26 +70,21 @@ public IInfoBar AttachInfoBar(Guid toolWindowGuid, string message, SonarLintImag
...
         public void DetachInfoBar(IInfoBar currentInfoBar)
         {
             if (currentInfoBar == null)
             {
                 throw new ArgumentNullException(nameof(currentInfoBar));
             }
 
-            PrivateInfoBarWrapper wrapper = currentInfoBar as PrivateInfoBarWrapper;
-            if (wrapper == null)
+            if (!(currentInfoBar is PrivateInfoBarWrapper wrapper))
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/InfoBar/InfoBarManager.cs:80]: Miscellaneous: code smell

File: src/Integration.Vsix/InfoBar/InfoBarManager.cs [hunk 3]
```diff
@@ -70,26 +70,21 @@ public IInfoBar AttachInfoBar(Guid toolWindowGuid, string message, SonarLintImag
...
             {
                 throw new ArgumentException(Strings.InvalidInfoBarInstance, nameof(currentInfoBar));
             }
 
             currentInfoBar.Close();
 
+            ThreadHelper.ThrowIfNotOnUIThread();
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/InfoBar/InfoBarManager.cs:87]: There are two possible paths through the code and I've added the thread helper checker in both places.
However, it hasn't stopped the VSX analyzer complaining, though - there are still half-a-dozen other analysis issues in this file, even though they are in lower-level methods that are guarded by a check in an upstream method.

**Comment by duncanp-sonar** [comment]
I've manually tested that the "ruleset out of date" gold bar still works correctly i.e. doesn't throw.

File: src/Integration.Vsix/InfoBar/InfoBarManager.cs (hunk 3 above)
Comment (rita-g-sonarsource) [reply src/Integration.Vsix/InfoBar/InfoBarManager.cs:87]: Is the `currentInfoBar.Close();` safe to run on non-UI thread?

**Review by rita-g-sonarsource — APPROVED** [review]
Just a question about IVsInfoBarUIElement.Close running on non-UI thread

**Pull request merged**